### DISCUSSION TITLE
Solver answering with unevaluated expression

### DISCUSSION THREAD

**Discussion by pfroehle-ccl** [cvc5/cvc5#10766] [Q&A] [answered]
Hello all,

for below input cvc5 v1.1.2 answers with gibberish.

Input: in.smt2

```smt2
(set-logic QF_NRA)

(declare-fun var () Real)
(assert (= (* (* (* var var) var) var) (/ 5 1)))
(check-sat)
(get-model)
```

Cmd: cvc5 --produc-models in.smt2

Output:

```smt2
sat
(
(define-fun var () Real (_ real_algebraic_number <1*x^4 + (-5), (5/4, 3/2)>))
)
```

Why is that?
Is there any way to have the solver answer explicitly?

Thanks!

**Comment by martin-cs** [comment, marked as answer]
The problem you have posed is var^4 = 5.  So the solution is when var is a 4th root of 5.  There are two real roots of this form which are +/- 1.4953...

The problem is that there isn't a way of expressing these directly in SMT-LIB.  So instead the solver tells you there is a solution (`sat`) and that that it is when `var` is a root of `1*x^4 + (-5)` (i.e. the exact answer) and that this is in the range `[1.25, 1.5]` (an approximate answer).  I feel this is an explicit answer.  What kind of answer were you expecting to get?

**Reply by ghost** [reply]
This seems as explicit as it can get.
However, for `var^3=5` theres exactly one result.

```smt2
(set-logic QF_NRA)

(declare-fun var () Real)
(assert (= (* (* var var) var) (/ 5 1)))
(check-sat)
(get-model)
```

Yet, the solver answers like-wise.

```smt2
sat
(
(define-fun var () Real (_ real_algebraic_number <1*x^3 + (-5), (3/2, 7/4)>))
)
```

Why does it answer with an interval again?

**Reply by martin-cs** [reply]
Because different people have different views on what "the answer" should be.  Some want it in exact, algebraic form.  Some want it in a numerical form.  So we give both.
